### ISSUE TITLE
Listener not updated if props change

### ISSUE THREAD

**Issue by mikekellyio** [green-arrow/react-firestore#9]
* `react-firestore` version: 0.6.0
* `node` version: 8.9.0
* `npm` (or `yarn`) version: 5.5.1

Relevant code or config

```jsx
<FirestoreCollection
  path="leagues"
  render={({ isLoading, data }) => {
    return isLoading ? (
      <Loading />
    ) : (
      <div className="form-group">
        <label htmlFor="leagueId">League</label>
        <select
          className="form-control"
          name="leagueId"
          value={this.state.leagueId}
          onChange={this.updateVal}
        >
          <option value="">Please select a league...</option>
          {data &&
            data.map(league => (
              <option key={league.id} value={league.id}>
                {league.name}
              </option>
            ))}
        </select>
      </div>
    );
  }}
/>

<FirestoreCollection
  path="teams"
  filter={["leagueId", "==", this.state.leagueId]}
  render={({ isLoading, data }) => {
    return isLoading ? (
      <Loading />
    ) : (
      <div className="form-group">
        <label htmlFor="teamId">Team</label>
        <select
          className="form-control"
          name="teamId"
          value={this.state.teamId}
          onChange={this.updateVal}
        >
          <option value="">Please select a team...</option>
          {data &&
            data.map(team => (
              <option key={team.id} value={team.id}>
                {team.name}
              </option>
            ))}
        </select>
      </div>
    );
  }}
/>

```

What you did:
I have 2 form elements. When the user selects a league, I want the team select to update with that particular leagues teams. 

What happened:
The team dropdown never updates to reflect leagueId changes because the `FirestoreCollection` listener is only set up on mount, instead of prop change.

Reproduction repository:

Problem description:

Suggested solution:
`FirestoreCollection` and `FirestoreDocument` should be refactored to update on prop change.

**Comment by damonbauer** [comment]
Another use case:

I've got a container component that mounts  a `FirestoreDocument` and renders a stateless component with the `data` received from `FirestoreDocument`.

Using `react-router`, I have a link to a route that mounts said container. If I'm already on a route that has the container mounted, I never getting a request for new data. Here's an example:

While at route: `example.com/item/itemID1`, I'm mounting this:
```jsx
<FirestoreDocument
  path={`items/${id}`} // `id` will be "itemID1"
  render={({ isLoading, data }) =>
    isLoading ? <Loading /> : <ItemDetail {...data} />
  }
/>
```
If I click a link to `example.com/item/itemID2`, the `path` prop in the `FirestoreDocument` _does_ get updated with `itemID2`, but it does not kick off a request to fetch that data; nothing changes, so my `ItemDetail` component never re-renders with any new data.

**Comment by mikekellyio** [comment]
@damonbauer I had some trouble with that as well, but was able to get it to work using:
```jsx
<Route
                path="/leagues/:id"
                render={props => (
                  <FirestoreDocument
                    path={"/leagues/" + props.match.params.id}
                    render={({ isLoading, data }) => {
                      return isLoading ? (
                        <Loading />
                      ) : (
                        <League league={data} {...props} {...this.props} />
                      );
                    }}
                  />
                )}
              />
```
I believe this is working because by using the render option for `Route` the `FirestoreDocument` is getting completely unmounted/remounted each time the route fires.

**Comment by damonbauer** [comment]
Closed by #10
